javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,106 @@
+const myObj = {};
+const todoArr = [];
+const myMonth = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+
+
+// Voice assistant function
+function getReply(command) {
+
+    if (command) {
+        const commandArr = command.split(" ");
+        command = command.toLowerCase();
+
+        // Get user name
+        if (command.startsWith("hey") || command.startsWith("hello") || command.startsWith("hi") || command.startsWith("my name")) {
+            let word = commandArr.pop();
+            if (Object.keys(myObj).length === 0) {
+                myObj.name = word;
+            }
+            return `Nice to meet you ${myObj.name}`;
+
+        } else if (command.startsWith("what") && command.endsWith("name?")) {
+            if (myObj.name) {
+                return `Your name is ${myObj.name}`;
+            } else {
+                return "You have not mentioned your name, what is your name?";
+            }
+
+        } else if (command.endsWith("my name")) {
+            word = commandArr.shift();
+            if (Object.keys(myObj).length === 0) {
+                myObj.name = word;
+            }
+            return `Nice to meet you ${myObj.name}`;
+
+        // Create a todo list
+        } else if (command.includes("add") && (command.includes("singing") || command.includes("fishing"))) {
+            word = commandArr[1];
+            if (word === "singing") {
+                word += " in the shower";
+            }
+            todoArr.push(word)
+            myObj.todos = todoArr;
+            return `${word} added to your todo`;
+
+        } else if (command.includes("what") && command.endsWith("todo?")) {
+            word = todoArr.join(" and ");
+            return `you have ${todoArr.length} todos - ${word}`;
+
+        // Remove fishing from the todo list
+        } else if (command.includes("remove") && command.includes("fishing")) {
+
+            if (todoArr.includes("fishing")) {
+                let index = todoArr.indexOf("fishing");
+                word = todoArr.splice(index, 1);
+                return `Removed ${word} from your todo`;
+            }
+            return "Fishing is not in your todo";
+
+        // Get today's date
+        } else if (command.startsWith("what") && command.includes("today")) {
+            let myDate = new Date();
+            return `${myDate.getDate()} of ${myMonth[myDate.getMonth()]} ${myDate.getFullYear()}`;
+
+        // Perform arithmetic operations
+        } else if (command.includes("*") || command.includes("+") || command.includes("-") || command.includes("/")) {
+            word = commandArr.slice(-3).join(' ');
+            return `${eval(word)}`;
```

Comment:
**note** that's correct and fine  for study project. 
But in real work `eval` is not recommended to use, it's dangerous.
e.g. https://www.geeksforgeeks.org/is-javascripts-eval-evil/